src/components/artwork_grid.tsx

```diff
@@ -3,14 +3,29 @@ import * as Relay from "react-relay"
 import styled from "styled-components"
 import Artwork from "./artwork/index"
 
-export interface GridProps extends RelayProps, React.HTMLProps<ArtworkGrid> {
+interface Props extends RelayProps, React.HTMLProps<ArtworkGrid> {
   columnCount?: number,
   sectionMargin?: number,
   itemMargin?: number,
+  onLoadMore?: any,
 }
 
-export class ArtworkGrid extends React.Component<GridProps, null> {
-  public static defaultProps: Partial<GridProps>
+interface State {
+  loading: boolean,
+}
+
+export class ArtworkGrid extends React.Component<Props, State> {
+  public static defaultProps: Partial<Props>
+
+  componentDidMount() {
+    window.onscroll = () => {
```

Comment:
Also, the creator of jQuery suggests not putting actual jobs into the window scroll event: http://ejohn.org/blog/learning-from-twitter/